### ISSUE TITLE
Incorporate the Windows Subsystem for Linux

### ISSUE THREAD

**Issue by mattwojo** [files-community/Files#295]
**Problem:**
The Windows Subsystem for Linux is extremely popular among developers, but for many, especially newer devs, the concept of switching between the Windows file system and Linux file system is very confusing. Currently, the Linux file system can be opened (after installing WSL and a distro) using Windows File Explorer using 'explorer.exe .'    ...but it would be VERY helpful to have a separation in the File Explorer, similar to how you've separated OneDrive, that visually indicates to the user whether they are currently in the Windows or Linux File System... and which Linux distros file system they are in (Ubuntu? Debian? SUSE?)  This would be HUGE!

**Describe the solution you'd like**

![image](https://user-images.githubusercontent.com/7566797/68800873-bb9b1380-060f-11ea-964d-7bae9d2d3df0.png)

**Comment by mattwojo** [comment]
Visual Studio Code has implemented a similar sort of thing with their Extensions installation windows...

![image](https://user-images.githubusercontent.com/7566797/68802417-c0ad9200-0612-11ea-8967-965266493308.png)

**Comment by mattwojo** [comment]
This is a clear visual indicator letting users know whether they've installed an extension on the local (Windows) drive, or on the Windows Subsystem for Linux (Ubuntu-18.04 in this case) drive... as well as an indicator in the lower-left corner of VS Code letting the user know what system their current project is running on.  
The community has LOVED this new remote support... definitely something that should be in File Explorer IMO and would be super cool to see it envisioned in a project like this.

**Comment by yaichenbaum** [comment]
This is a great idea! One of the things we are looking to do with this project is add value and I definitely think this fits under that.

**Comment by duke7553** [comment]
I plan to investigate this feature during the v0.7 cycle.

**Comment by duke7553** [comment]
The v0.7 cycle has seen lots of work go into this, but it hasn't been fully tested/finalized yet, so I'm bumping the target for WSL support to v0.8

**Comment by lampenlampen** [comment]
Actually you can already access your WSL-files in files with `\\wsl$`. This only works for now, if the distro is running ([see](https://devblogs.microsoft.com/commandline/whats-new-for-wsl-in-windows-10-version-1903/) under Known Issues).

So we only need a way to detect which distros are installed, then we can display them on the sidebar. The files in a distro can be accessed with `\\wsl$\{distro_name}`. 

I think we can even go a step further and start the distro for the user. We only have to remember to close it at app termination.

**Comment by duke7553** [comment]
@lampenlampen Yes! I actually started work on this last year. If you want to improve upon it, and hook it up, take a look at this: https://github.com/duke7553/files-uwp/blob/bf942034ef2b7e11a6f6ccf4183b64a6e17716eb/Files/View%20Models/SettingsViewModel.cs#L174

However, I do plan on working on this myself very soon.

**Comment by SketchNI** [comment]
> I think we can even go a step further and start the distro for the user. We only have to remember to close it at app termination.

I wouldn't close out the distro when Files closes.
I'd instead add a button to the listing so something like
```
Ubuntu 20.04    ⏹️
openSUSE Leap        
Kali                            
```
A standby-style icon would best denote it but that isn't an emoji.
You could go one step further and allow something like 

```
Ubuntu 20.04    ⏹️
openSUSE Leap   ▶️
Kali            ▶️
```
to allow WSl distros to be started via Files as well as terminated. That may be beyond the scope of the project but could be added under a "EnableDeveloperFriendlyFeatures"

**Comment by DiegoMagdaleno** [comment]
Hey I want to report something else, and I think you mean you only want to start the WSL kernel.

Example:
See I dont have any sessions of WSL running, while I started Pengwin distro earlier, for example I didn't touch Void Linux at all Today.

Here is an Example, i dont have any WSL sessions open:
![image](https://user-images.githubusercontent.com/38844659/101953800-a4d34180-3bc0-11eb-99dc-a869dcb58259.png)

However I can access them just fine 
![image](https://user-images.githubusercontent.com/38844659/101953844-b9afd500-3bc0-11eb-9356-ba24f47d3284.png)

I don't think Files should start the WSL distros, as they all start once the kernel is started.

I dont know if maybe is because I am on WSL2

**Comment by tomocrafter** [comment]
You can open the wsl directories on this app, but if you press the reload button then all of entries will be duplicated like this (but this is just for visual, any of those isn't duplicated if you check it on wsl).
![image](https://user-images.githubusercontent.com/10905364/104751541-e2ffea00-5798-11eb-9480-d304f09cb661.png)

**Comment by yaichenbaum** [comment]
@tomocrafter Can you check if you are running v1.1??

**Comment by tomocrafter** [comment]
@yaichenbaum I'm sorry I tested it on v1.1 and it couldn't be reproduced. Thanks for fixing it.

**Comment by yaichenbaum** [comment]
@tomocrafter Thank you for the update!

**Comment by yaichenbaum** [comment]
@mattwojo This feature was implemented with #3782.
